enterprise/server/backends/pebble_cache/pebble_cache.go

```diff
@@ -315,6 +330,70 @@ func partitionLimits(partID string) ([]byte, []byte) {
 	return keyRange(key)
 }
 
+func batchEditAtime(batch *pebble.Batch, fileMetadataKey []byte, fileMetadata *rfpb.FileMetadata) error {
+	atime := time.UnixMicro(fileMetadata.GetLastAccessUsec())
+	age := time.Since(atime)
+	if age < *atimeUpdateThreshold {
+		return nil
+	}
+	fileMetadata.LastAccessUsec = time.Now().UnixMicro()
+	protoBytes, err := proto.Marshal(fileMetadata)
+	if err != nil {
+		return err
+	}
+	return batch.Set(fileMetadataKey, protoBytes, nil /*ignored write options*/)
+}
+
+func (p *PebbleCache) processAccessTimeUpdates(quitChan chan struct{}) {
+	batch := p.db.NewBatch()
+
+	flush := func() error {
+		if batch.Count() > 0 {
+			if err := batch.Commit(&pebble.WriteOptions{Sync: true}); err != nil {
+				return err
+			}
+			batch = p.db.NewBatch()
+		}
+		return nil
+	}
+
+	exitMu := sync.Mutex{}
+	exiting := false
+	go func() {
+		<-quitChan
+		exitMu.Lock()
+		exiting = true
+		exitMu.Unlock()
+	}()
+
+	for {
+		select {
+		case accessTimeUpdate := <-p.accesses:
+			md, err := readFileMetadata(p.db, accessTimeUpdate.fileMetadataKey)
+			if err != nil {
+				log.Warningf("Error unmarshaling data for %q: %s", accessTimeUpdate.fileMetadataKey, err)
+				continue
+			}
+			if err := batchEditAtime(batch, accessTimeUpdate.fileMetadataKey, md); err != nil {
+				log.Warningf("Error updating atime: %s", err)
+			}
+			if int(batch.Count()) >= *atimeWriteBatchSize {
+				flush()
+			}
+		case <-time.After(atimeFlushPeriod):
+			flush()
+		case <-time.After(time.Second):
+			exitMu.Lock()
+			done := exiting
```

Comment:
Why not add a case for quitChan directly instead of this roundabout check?

Reply:
Doing it this way allows us to drain the atime queue before which I wanted